resources/session02/http_server.py

```diff
@@ -1,12 +1,73 @@
 import socket
 import sys
+import mimetypes
+import os.path
+
+
+def response_ok(body=b"this is a pretty minimal response", mimetype=b"text/plain"):
+    """returns a basic HTTP response"""
+    mimetype = b"content-type:" + mimetype
+    resp = []
+    resp.append(b"HTTP/1.1 200 OK")
+    resp.append(mimetype)
+    resp.append(b"")
+    resp.append(body)
+    return b"\r\n".join(resp)
+
+
+def response_method_not_allowed():
+    """returns a 405 Method Not Allowed response"""
+    resp = []
+    resp.append("HTTP/1.1 405 Method Not Allowed")
+    resp.append("")
+    return "\r\n".join(resp).encode('utf8')
+
+
+def response_not_found():
+    """returns a 404 Not Found response"""
+    return b"HTTP/1.1 404 Not Found"
```

Comment:
This is a good first line, but is not a complete HTTP response.  Remember that to be complete, it must have the empty line that marks the division between the headers and the body.